src/blockdata/script.rs

```diff
@@ -363,6 +389,58 @@ impl Script {
         self.is_cp2pkh() || self.is_cp2sh()
     }
 
+    /// Check if a script pubkey is multisig script
+    pub fn is_multisig(&self) -> bool {
+        if self.0.len() < 4 || *self.0.last().unwrap() != opcodes::all::OP_CHECKMULTISIG.into_u8() {
+            return false;
+        }
+        let required_count = match opcodes::All::from(self.0[0]).classify() {
+            opcodes::Class::PushNum(i) => i,
+            _ => { return false }
+        };
+        let pubkey_count = match opcodes::All::from(self.0[self.0.len() - 2]).classify() {
+            opcodes::Class::PushNum(i) => i,
+            _ => { return false }
+        };
+        required_count <= pubkey_count
+    }
+
+    /// Return public keys for multisig script, and the number of signatures required
+    pub fn get_multisig_pubkeys(&self) -> Result<(i32, Vec<PublicKey>), MultisigError> {
+        if !self.is_multisig() {
+            return Err(MultisigError::IsNotMultisig);
+        }
```

Comment:
With this check, all later `InvalidRequiredSigCount` and `InvalidPublicKey` errors are also `IsNotMultisig`.

Reply:
Remove `InvalidRequiredSigCount` and `InvalidPublicKey`